projects/train/train/data/supervised/frequency_domain.py

```diff
@@ -1,24 +1,24 @@
-from ml4gw.transforms.spectrogram import MultiResolutionSpectrogram
+import torch
+
+from ml4gw.transforms.qtransform import SingleQTransform
 from projects.train.train.data.base import Tensor
-from train.data.supervised import SupervisedAframeDataset
+from train.data.supervised.supervised import SupervisedAframeDataset
 
 
 class FrequencyDomainSupervisedAframeDataset(SupervisedAframeDataset):
-    def __init__(
-        self, *args, spectrogram: MultiResolutionSpectrogram, **kwargs
-    ):
+    def __init__(self, *args, qtransform: SingleQTransform, **kwargs):
         super().__init__(*args, **kwargs)
-        self.spectrogram = spectrogram
+        self.qtransform = qtransform
 
     def augment(self, X):
         X, y = super().augment(X)
-        X = self.spectrogram(X)
+        X = self.qtransform(X)
         return X, y
 
     def build_val_batches(
         self, background: Tensor, signals: Tensor
     ) -> tuple[Tensor, Tensor]:
         X_bg, X_fg = super().build_val_batches(background, signals)
-        X_bg = self.spectrogram(X_bg)
-        X_fg = self.spectrogram(X_fg)
+        X_bg = self.qtransform(X_bg)
+        X_fg = torch.stack([self.qtransform(X) for X in X_fg])
```

Comment:
Can you explain why this is need here?

Reply:
Because `X_fg` is [4D tensor](https://github.com/ML4GW/aframev2/blob/380921c48bf593a41e81298f4b7abd3b84f60c0a/projects/train/train/data/base.py#L467), and the `qtransform` requires 3 or fewer dimensions. This is what I was trying to solve with that `ml4gw` PR, but the interpolation issues don't seem to permit any nice solutions.